mkosi/machine.py

```diff
@@ -62,19 +63,19 @@ def serial(self) -> pexpect.spawn:
             raise ValueError(
                         dedent(
                             """\
-                            Trying to access serial console before machine boot or after machine shutdown.
+                            Trying to access serial console before machine boots or after machine shutdown.
                             In order to boot the machine properly, use it as a context manager.
                             Then, a Mkosi image will be booted in the __enter__ method.
                             """
                         )
                     )
         return self._serial
 
-    def ensure_booted(self) -> None:
+    def __ensure_booted(self) -> None:
         # Try to access the serial console which will raise an exception if the machine is not currently booted.
         assert self._serial is not None
 
-    def __enter__(self) -> Machine:
+    def __build_image(self) -> None:
```

Comment:
This doesn't really need to be a separate method, we can just keep this logic in __enter__